### DISCUSSION TITLE
should it be 'static instead '_ ?

### DISCUSSION THREAD

**Discussion by AlexanderARodin** [emilk/egui#3793] [Q&A]
https://github.com/emilk/egui/blob/401de056306d632b1082247e152c93479d88aa20/crates/egui_demo_lib/src/demo/toggle_switch.rs#L105C54-L105C55

I'm trying to create custom widget and Rust complains about life-time.
I'm not very hard-core Rustacean, therefore I just tried to read compiler recommendation and it works. But I'm not sure, it's correct way for **'static** instead of original **'_**

Thank you in advance,
A.

**Comment by ItsEthra** [comment]
Please provide your code

**Reply by AlexanderARodin** [reply]
I tried to repeat example for custom-widget source example:
```
pub fn toggle(on: &mut bool) -> impl egui::Widget + '_ {
    move |ui: &mut egui::Ui| toggle_ui(ui, on)
}
```
Rust doesn't compile it in source variant and complains on life-time. As far I know at the moment, there is no such a life-time like **'_**.
So I changed it to **'static**:
```
pub fn toggle(on: &mut bool) -> impl egui::Widget + 'static {
    move |ui: &mut egui::Ui| toggle_ui(ui, on)
}
```
It worked.

On the other hand, I've just tried to delete life-time marker at all... And it works too.
I don't know (yet) why the source Example works. But now I'm finally sure it's just my lack of deeper understanding life-times.

Anyway, thank you for answer.
In current state it seems I have no subject for discussion.
I'm sorry. The thread can be deleted.
